app/services/discord_messages/creator.rb

```diff
@@ -0,0 +1,21 @@
+module DiscordMessages
+  class Creator
+    attr_reader :message
+
+    def initialize(params:)
+      @params = params
+    end
+
+    def call
+      @message = DiscordMessage.new(channel_id: params['channel_id'], external_id: params['id'])
+
+      @message.save!
+    rescue ActiveRecord::RecordInvalid => e
+      raise CreationFailed, e.record.errors.full_messages.join(', ')
```

Comment:
I don't think `CreationFailed` exists. Perhaps you meant `Messages::CreationFailed`?

```suggestion
      raise Messages::CreationFailed, e.record.errors.full_messages.join(', ')
```

Reply:
Done!